scripts/rabbitmq-server-ha.ocf

```diff
@@ -260,6 +262,16 @@ ${OCF_RESKEY_binary} should listen on this port
 <content type="boolean" default="${OCF_RESKEY_node_port_default}" />
 </parameter>
 
+<parameter name="vhost" unique="0" required="0">
+<longdesc lang="en">
+Main virtual host used in the cluster, guaranteed to not suffer data loss. Other
+virtual hosts may exist, but some scenarios can lead to data loss for them. This
```

Comment:
I think that you're putting too much hope in what this script can do. It just strives to minimize the data loss by giving a broker some time to finish queue synchronization before taking some destructive actions. But there is no guarantees at all that data won't be lost - i.e. in some worst case scenario can lose all the data. And in this worst case scenario you'll **lose** not only all the messages, but also **all custom vhosts** themselves - only `default_vhost` from `rabbitmq.config`  will exist afterwards.

This is the price that you are paying for the fact that OCF script will **automatically** recover cluster to a working state from any starting state.